src/storage/src/compaction/twcs.rs

```diff
@@ -0,0 +1,391 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::fmt::{Debug, Formatter};
+use std::marker::PhantomData;
+
+use common_telemetry::tracing::warn;
+use common_telemetry::{debug, info};
+use common_time::timestamp::TimeUnit;
+use common_time::timestamp_millis::BucketAligned;
+use common_time::Timestamp;
+use store_api::logstore::LogStore;
+
+use crate::compaction::picker::get_expired_ssts;
+use crate::compaction::task::CompactionOutput;
+use crate::compaction::{infer_time_bucket, CompactionRequestImpl, CompactionTaskImpl, Picker};
+use crate::sst::{FileHandle, LevelMeta};
+
+pub struct TwcsPicker<S> {
+    max_files_in_active_window: usize,
```

Comment:
I think we should make these options to be the table options such as `compact.twcs.max_files_in_active_window` and `compact.twcs.max_files_in_non_active_window`.